### PR TITLE
Distro version improvements

### PR DESCRIPTION
**Pull request by scop**

### CODE REVIEW

**Comment by scop** [comment]
The new validator will in addition to [ArchLinux] and [Debian] be used in [Fedora] and [Ubuntu].

File: services/debian/debian.service.js [hunk 1]
```diff
@@ -47,7 +47,14 @@ module.exports = class Debian extends BaseJsonService {
...
       schema,
-      url: `https://api.ftp-master.debian.org/madison?f=json&s=${distribution}&package=${packageName}`,
+      url: 'https://api.ftp-master.debian.org/madison',
+      options: {
+        qs: {
+          f: 'json',
+          s: distribution,
+          package: packageName,
+        },
+      },
```
Comment (paulmelnikow) [inline services/debian/debian.service.js]: 👍

File: services/debian/debian.tester.js [hunk 2]
```diff
@@ -1,15 +1,22 @@
...
     label: 'debian',
-    message: isVPlusDottedVersionNClausesWithOptionalSuffix,
+    message: isVPlusDottedVersionNClausesWithOptionalSuffixAndEpoch,
+  })
+
+t.create('Debian package (default distribution, valid, query unsafe chars)')
+  .get('/g++.json')
+  .expectBadge({
+    label: 'debian',
+    message: isVPlusDottedVersionNClausesWithOptionalSuffixAndEpoch,
```
Comment (paulmelnikow) [inline services/debian/debian.tester.js]: 👍

File: services/test-validators.js [hunk 3]
```diff
@@ -26,6 +26,12 @@ const isVPlusDottedVersionNClausesWithOptionalSuffix = withRegex(
   /^v\d+(\.\d+)*([-+~].*)?$/
 )
 
+// same as above, but also accepts an optional 'epoch' prefix that can be
+// found e.g. in distro package versions
```
Comment (paulmelnikow) [inline services/test-validators.js]: Could you include in this comment an example of the kind of string you want to match? I think that will help people who are unfamiliar with these distro package versions.

Comment (scop) [reply services/test-validators.js]: Done

File: services/test-validators.js [hunk 4]
```diff
@@ -27,7 +27,7 @@ const isVPlusDottedVersionNClausesWithOptionalSuffix = withRegex(
 )
 
 // same as above, but also accepts an optional 'epoch' prefix that can be
-// found e.g. in distro package versions
+// found e.g. in distro package versions, like 4:6.3.0-4
```
Comment (paulmelnikow) [inline services/test-validators.js]: Thanks!

**Pull request merged**